src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs

```diff
@@ -143,17 +145,55 @@
+                                               :value (get (:attributes @state) property-kwd nil)
+                                               :onChange update-property
+                                               })
                      :else
                      (style/create-text-field {:style (colorize {:width "100%"})
                                                :type (cond (= renderHint "date") "date"
                                                            (= type "integer") "number"
                                                            :else "text")
                                                :min minimum
+                                               :disabled disabled?
                                                :placeholder inputHint
                                                :value (get (:attributes @state) property-kwd)
-                                               :onChange update-property}))]))
-          questions)]))})
 
+                                               :onChange update-property}))]))
+          questions)]))
+   :component-did-mount
+   (fn [{:keys [props state refs]}]
+     (let [options (js/Bloodhound.
+                     (clj->js
+                       {:datumTokenizer js/Bloodhound.tokenizers.whitespace
+                        :queryTokenizer js/Bloodhound.tokenizers.whitespace
+                        :remote (clj->js {:url (str (config/api-url-root) "/duos/autocomplete/%QUERY")
+                                          :wildcard "%QUERY"})}))
+           existing-ontology (get-in props [:workspace :workspace :library-attributes :library:diseaseOntologyLabel])
+           property-kwd (@refs "library:diseaseOntologyLabel")] ;; fix this
+       (.typeahead (js/$ property-kwd) ;;
+                   (clj->js {:highlight true
+                             :hint true
+                             :minLength 1})
+                   (clj->js
+                     {:source options
+                      :display (fn [result]
+                                 (aget result "label"))
+                      :templates (clj->js
+                                   {:empty "<div> unable to find any matches to the current query </div>"
+                                    :suggestion
+                                    (fn [result]
+                                      (str "<div> <div style='font-weight: bold; line-height: 1.5em;'>" (aget result "label")
+                                           "</div><div>" (aget result "id") "</div>"
+                                           "<div style='font-size: small; font-style: italic'> " (aget result "definition") "</div></div>"))})}))
+       (.bind (js/$ property-kwd)
+              "typeahead:select"
+              (fn [ev suggestion]
+                (swap! state update :attributes assoc :library:diseaseOntologyLabel (aget suggestion "label") :library:diseaseOntologyID (aget suggestion "id"))))))})
```

Comment:
It would be nice if this function were not specific to catalog and instead put into endpoints or even it's own component file so we can re-use it. I think it could be more generic that way too ... pass in the search box binding, an autocomplete term, and an endpoint and let the function do all the work in returning content.